tools/compile-keymap.c

```diff
@@ -114,29 +138,59 @@ parse_options(int argc, char **argv, struct xkb_rule_names *names)
         OPT_MODEL,
         OPT_LAYOUT,
         OPT_VARIANT,
-        OPT_OPTION,
+        OPT_OPTION
     };
-    static struct option opts[] = {
-        {"help",             no_argument,            0, 'h'},
-        {"verbose",          no_argument,            0, OPT_VERBOSE},
+    int warnings = XKB_WARNING_NONE;
+    struct option opts[] = {
+        {"help",             no_argument,            NULL, 'h'},
+        {"verbose",          no_argument,            NULL, OPT_VERBOSE},
 #if ENABLE_PRIVATE_APIS
-        {"kccgst",           no_argument,            0, OPT_KCCGST},
+        {"kccgst",           no_argument,            NULL, OPT_KCCGST},
 #endif
-        {"rmlvo",            no_argument,            0, OPT_RMLVO},
-        {"from-xkb",         no_argument,            0, OPT_FROM_XKB},
-        {"include",          required_argument,      0, OPT_INCLUDE},
-        {"include-defaults", no_argument,            0, OPT_INCLUDE_DEFAULTS},
-        {"rules",            required_argument,      0, OPT_RULES},
-        {"model",            required_argument,      0, OPT_MODEL},
-        {"layout",           required_argument,      0, OPT_LAYOUT},
-        {"variant",          required_argument,      0, OPT_VARIANT},
-        {"options",          required_argument,      0, OPT_OPTION},
+        {"rmlvo",            no_argument,            NULL, OPT_RMLVO},
+        {"from-xkb",         no_argument,            NULL, OPT_FROM_XKB},
+        {"include",          required_argument,      NULL, OPT_INCLUDE},
+        {"include-defaults", no_argument,            NULL, OPT_INCLUDE_DEFAULTS},
+        {"rules",            required_argument,      NULL, OPT_RULES},
+        {"model",            required_argument,      NULL, OPT_MODEL},
+        {"layout",           required_argument,      NULL, OPT_LAYOUT},
+        {"variant",          required_argument,      NULL, OPT_VARIANT},
+        {"options",          required_argument,      NULL, OPT_OPTION},
+        // Warnings
+        {"Wunrecognized-keysym",                  no_argument, &warnings, XKB_WARNING_UNRECOGNIZED_KEYSYM},
```

Comment:
this is a long option, so it'd be `--Wunrecognized-keysym` - unlike the gcc option. That seems a bit odd, given that the format is supposed to resemble the compiler options.

Instead, you could do it like `c = getopt_long(argc, argv, "hW:", opts, &option_index);` and the `optarg` to `'W'` is the flag. You'll have to strcmp instead of having it in this option then but I don't think that's a huge issue. Plus it'd be easy to add the `E` handling (see below) then with the same code if you factor it out into a helper function.

Reply:
Fixed